src/platform/plugins/shared/discover/public/embeddable/get_search_embeddable_factory.test.tsx

```diff
@@ -14,35 +14,62 @@ import { createSearchSourceMock } from '@kbn/data-plugin/public/mocks';
 import type { DataView } from '@kbn/data-views-plugin/common';
 import { SHOW_FIELD_STATISTICS } from '@kbn/discover-utils';
 import { buildDataViewMock, deepMockedFields } from '@kbn/discover-utils/src/__mocks__';
-import type { BuildReactEmbeddableApiRegistration } from '@kbn/embeddable-plugin/public/react_embeddable_system/types';
 import type { PresentationContainer } from '@kbn/presentation-containers';
-import type {
-  PhaseEvent,
-  PublishesUnifiedSearch,
-  StateComparators,
-} from '@kbn/presentation-publishing';
+import type { PhaseEvent, PublishesUnifiedSearch } from '@kbn/presentation-publishing';
 import { VIEW_MODE } from '@kbn/saved-search-plugin/common';
 import { act, render, waitFor } from '@testing-library/react';
 
 import type { AggregateQuery, Filter, Query, TimeRange } from '@kbn/es-query';
 import { createDataViewDataSource } from '../../common/data_sources';
 import { discoverServiceMock } from '../__mocks__/services';
 import { getSearchEmbeddableFactory } from './get_search_embeddable_factory';
-import type {
-  SearchEmbeddableApi,
-  SearchEmbeddableRuntimeState,
-  SearchEmbeddableSerializedState,
-} from './types';
+import type { SearchEmbeddableApi, SearchEmbeddableRuntimeState } from './types';
+
+jest.mock('./utils/serialization_utils', () => ({}));
 
 describe('saved search embeddable', () => {
+  const dataViewMock = buildDataViewMock({ name: 'the-data-view', fields: deepMockedFields });
+
+  const getInitialRuntimeState = ({
+    searchMock,
+    dataView = dataViewMock,
+    partialState = {},
+  }: {
+    searchMock?: jest.Mock;
+    dataView?: DataView;
+    partialState?: Partial<SearchEmbeddableRuntimeState>;
+  } = {}): SearchEmbeddableRuntimeState => {
+    const searchSource = createSearchSourceMock({ index: dataView }, undefined, searchMock);
+    discoverServiceMock.data.search.searchSource.create = jest
+      .fn()
+      .mockResolvedValueOnce(searchSource);
+
+    return {
+      timeRange: { from: 'now-15m', to: 'now' },
+      columns: ['message', 'extension'],
+      rowHeight: 30,
+      headerRowHeight: 5,
+      rowsPerPage: 50,
+      sampleSize: 250,
+      serializedSearchSource: searchSource.getSerializedFields(),
+      ...partialState,
+    };
+  };
+
+  let runtimeState = getInitialRuntimeState();
+
+  beforeEach(() => {
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    require('./utils/serialization_utils').deserializeState = () => runtimeState;
```

Comment:
Nit: IMO it would be cleaner to use a top level `mockRuntimeState` variable that gets reset in `beforeEach`, and returned directly from the `serialization_utils` mock instead. Not necessary to change, just a note.